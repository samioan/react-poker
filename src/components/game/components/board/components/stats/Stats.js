import React from "react";

const Stats = ({ money, bid, strength, pot }) => (
  <div className="board-row">
    {money > -1 && <h2 className="textArea">Money: {money}</h2>}
    {bid > -1 && <h2 className="textArea">&nbsp; Bid: {bid} </h2>}
    {strength && <h2 className="textArea">&nbsp; {strength} </h2>}
    {pot > -1 && <h2 className="textArea">Pot: {pot} </h2>}
  </div>
);

export { Stats };
export default Stats;
