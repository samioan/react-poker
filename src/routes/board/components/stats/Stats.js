import React from "react";

import classes from "./Stats.module.css";

const Stats = ({ money, bid, strength, pot }) => (
  <div className={classes.container}>
    {money > -1 && <h2 className={classes.text}>Money: {money}</h2>}
    {bid > -1 && <h2 className={classes.text}>Bid: {bid} </h2>}
    {strength && <h2 className={classes.text}>{strength} </h2>}
    {pot > -1 && <h2 className={classes.text}>Pot: {pot} </h2>}
  </div>
);

export { Stats };
export default Stats;
