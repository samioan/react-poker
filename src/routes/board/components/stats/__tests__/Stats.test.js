import { Stats } from "../Stats";

describe("Stats component", () => {
  test("renders correctly", () => {
    // GIVEN
    const mockProps = {
      money: 10,
      bid: 10,
      strength: "mock_strength",
      pot: 10,
    };

    // WHEN
    const tree = shallowSnapshot(<Stats {...mockProps} />);

    // THEN
    expect(tree).toMatchSnapshot();
  });
});
