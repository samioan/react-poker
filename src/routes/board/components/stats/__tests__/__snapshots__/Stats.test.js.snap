// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component renders correctly 1`] = `
<div
  className="board-row"
>
  <h2
    className="textArea"
  >
    Money: 
    10
  </h2>
  <h2
    className="textArea"
  >
      Bid: 
    10
     
  </h2>
  <h2
    className="textArea"
  >
      
    mock_strength
     
  </h2>
  <h2
    className="textArea"
  >
    Pot: 
    10
     
  </h2>
</div>
`;
